clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt

```diff
@@ -96,10 +94,10 @@ interface FossIdRestService {
     suspend fun checkScanStatus(@Body body: PostRequestBody): EntityPostResponseBody<ScanStatus>
 
     @POST("api.php")
-    suspend fun listScanResults(@Body body: PostRequestBody): MapResponseBody<FossIdScanResult>
+    suspend fun listScanResults(@Body body: PostRequestBody): EntityPostResponseBody<Any>
```

Comment:
Hmm. I actually like the original code here better as it shows the concrete return type. Also in general, with the exception of line 340 in `FossId.kt`, this change to me does not really make the code more readable.

I believe what we should do instead is to modify in `Extensions.kt` the signature

    inline fun <reified T : Any> EntityPostResponseBody<Any>.toList(): List<T> =

to

    inline fun <reified T : Any> EntityPostResponseBody<*>.toList(): List<T> =

and only keep the changes to `FossId.kt` from this commit.

Reply:
Ha yes. I was also unhappy about the return type and this `<*>` was the missing bit.

Ok, we can do as suggested but then I would like to adapt in the same way the return type of several functions in `FossIdRestService`:
`listScansForProject`, `listMarkedAsIdentifiedFiles`, `listIgnoredFiles`, `listPendingFiles`.

I would do so in a separate additional commit. Would it be fine for you ?

Reply:
Yes, sure.

Reply:
BTW, here are some more minor improvements that just occurred to me: https://github.com/oss-review-toolkit/ort/pull/3923

Reply:
Ok, I did some more testing and I remember now why we went with this design of `EntityPostResponseBody<Any>`.
As explained in the `toList` function, the FossID server is not consistent. It usually returns a List or a Map, but if there is no entry, it returns `data:false`.
The `toList` function abstracts this behaviour, but we nevertheless need a Kotlin representation that could be all these types at once, to be able to deserialize them from JSON.
If we put for example `MapResponseBody<FossIdScanResult>`, it will work as long a the server returns a collection, but if it returns `data:false`, then an exception will be thrown during JSON deserialization, before even being able to call `toList`.
That's why we should in fact use `EntityPostResponseBody<Any>` in all these functions.
`listScanResults` and `listIdentifiedFiles` were using a `MapResponseBody`, because in all our tests we always had Identified files and Scan results, but to be 100% safe, we should use EntityPostResponseBody instead.

Reply:
> If we put for example `MapResponseBody<FossIdScanResult>`, it will work as long a the server returns a collection, but if it returns `data:false`, then an exception will be thrown during JSON deserialization, before even being able to call `toList`.

Could you please add a test for this (probably as a separate PR), to ensure whatever improvements we do here, we don't break that requirement?

Reply:
Can I do it in the current PR but in a separate commit ?

If we put `EntityPostResponseBody<Any>` everywhere, have a test provides a nice reasoning for the changes ?

Reply:
Ok, let's give it a try how it looks like.

Reply:
Ok I will do it.
Then in its current state, this PR is done or ?

Reply:
I pushed a test.
The test data contains all three kinds of return type handled by `toList`: `data: false`, list and map.

Reply:
Also I got an Idee: to avoid having these generic return types which are a pain for the user, maybe for these functions we can pull the `checkReponse` and the `toList` to the extension functions in `Extensions.kt`.
Therefore the extension functions would have 'concrete' return types.
If you are ok with this option, I'll do it in a separate PR.

Reply:
I'm curious how this would look like concretely and am looking forward to a follow-up PR.